### DISCUSSION TITLE
How to configure when model training ends?

### DISCUSSION THREAD

**Discussion by adamkgoldfarb** [explosion/spaCy#7465] [Help: Best practices]
I searched pretty thoroughly through the docs for an answer, but couldn't find one-- apologies if this has already been asked:

How does spacy determine when training ends for a given session? Is there a space in the config where I can adjust this?

Thanks!

**Comment by polm** [comment]
Good question, it may have been hard to find this in the docs because there are multiple settings that control when training ends.

- `max_epochs` sets the number of epochs, typically complete passes over the training data, to use for training
- `max_steps` sets the max number of iterations, steps, or batches to process
- `patience` is an early stopping setting; training will stop after `patience` steps with no improvement

These are [here](https://spacy.io/api/data-formats#config-training) in the docs. You can set all of these and the first one to be triggered will stop training. Typically you'd use one of `max_epochs` or `max_steps` with `patience` possibly triggering early stopping.

**Reply by adamkgoldfarb** [reply]
Incredibly helpful, thank you very much! These various settings make sense, and now it's clear how they interact with one another.
